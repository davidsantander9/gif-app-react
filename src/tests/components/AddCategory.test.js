import React from 'react';
import { shallow } from 'enzyme';
import AddCategory from '../../components/AddCategory';

describe('<AddCategory/>  test', () => {

    const setCategories = () => {};
    const wrapper = shallow(<AddCategory setCategories={ setCategories } />);

    test('should correctly display the component', () => {
        expect( wrapper ).toMatchSnapshot();
    });

    test('should change input text', () => {
        const input = wrapper.find('input');
        const value = 'Hola mundo'; 
        input.simulate('change', { target:{ value } });
        expect(  wrapper.find('p').text().trim()  ).toBe(value);
    })
    
    
});