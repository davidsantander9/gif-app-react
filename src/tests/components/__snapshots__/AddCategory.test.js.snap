// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AddCategory/>  test should correctly display the component 1`] = `
<form
  onSubmit={[Function]}
>
  <p>
     
     
     
  </p>
  <input
    onChange={[Function]}
    type="text"
    value=" "
  />
</form>
`;
